packages/frontend/src/hooks/pivotTable/pivotQueryResults.ts

```diff
@@ -116,20 +117,38 @@ export const pivotQueryResults = ({
 
         for (let nMetric = 0; nMetric < metrics.length; nMetric++) {
             const metric = metrics[nMetric];
-            const indexRowValues = [
-                ...indexDimensions.map((d) => row[d].value),
-                ...(pivotConfig.metricsAsRows ? [metric] : []),
-            ];
-            const headerRowValues = [
-                ...headerDimensions.map((d) => row[d].value),
-                ...(pivotConfig.metricsAsRows ? [] : [metric]),
-            ];
+
+            const indexRowValues: PivotValue[] = indexDimensions
+                .map<PivotValue>((d) => ({
+                    type: 'value',
+                    fieldId: metric.fieldId,
+                    value: row[d].value,
+                }))
+                .concat(
+                    pivotConfig.metricsAsRows
+                        ? [{ type: 'label', fieldId: metric.fieldId }]
+                        : [],
+                );
+
+            const headerRowValues: PivotValue[] = headerDimensions
+                .map<PivotValue>((d) => ({
+                    type: 'value',
+                    fieldId: metric.fieldId,
+                    value: row[d].value,
+                }))
+                .concat(
+                    pivotConfig.metricsAsRows
+                        ? []
+                        : [{ type: 'label', fieldId: metric.fieldId }],
+                );
 
             // Write the index values
             if (
                 setKeyIfNotExists(
                     rowIndices,
-                    indexRowValues.map((r) => r.raw),
+                    indexRowValues.map((l) =>
+                        l.type === 'value' ? String(l.value.raw) : l.fieldId,
```

Comment:
I'm not sure what is happening here, why is there a string conversion now?

Reply:
raw value is unknown, we can use formatted instead but I think string conversion will do the work